frontend/public/components/factory/details.tsx

```diff
@@ -90,8 +89,8 @@ export const DetailsPage = withFallback<DetailsPageProps>(({ pages = [], ...prop
       ...dynamicResourcePageExtensions
         .filter(
           (p) =>
-            referenceForExtensionModel(p.properties.model) ===
-            (kindObj ? referenceFor(kindObj) : props.kind),
+            p.properties.model.group === kindObj.apiGroup &&
+            p.properties.model.kind === kindObj.kind,
```

Comment:
when the model contains a version, you should be checking it for a match as well.